astropy/io/fits/hdu/base.py

```diff
@@ -457,7 +457,7 @@ def block_iter(nbytes):
         hdu._data_size = size + _pad_length(size)
 
         # Checksums are not checked on invalid HDU types
-        if checksum and checksum != 'remove' and isinstance(hdu, _ValidHDU):
+        if isinstance(hdu, _ValidHDU):
```

Comment:
Not really. IMO When we use `fits.open` with `checksum=False` is bascially doesn't touch whatever CHECKSUM/DATASUM there is. So if one has used False, he doesn't care about CHECKSUM keys but that doesn't mean that `self._checksum` shouldn't be set. It should be set to whatever value of CHECKSUM there is (None otherwise).